x-pack/plugins/licensing/public/plugin.test.ts

```diff
@@ -104,26 +104,26 @@ describe('licensing plugin', () => {
           Promise.resolve(licenseMock.createLicense({ license: { type: types.shift() } }))
         );
 
-        await plugin.setup(coreSetup);
-        const { refresh, license$ } = await plugin.start(coreStart);
-
-        let i = 0;
-        license$.subscribe((value) => {
-          i++;
-          if (i === 1) {
-            expect(value.type).toBe('basic');
-            refresh();
-          } else if (i === 2) {
-            expect(value.type).toBe('gold');
-            // since this is a synchronous subscription, we need to give the exhaustMap a chance
-            // to mark the subscription as complete before emitting another value on the Subject
-            process.nextTick(() => refresh());
-          } else if (i === 3) {
-            expect(value.type).toBe('platinum');
-            done();
-          } else {
-            throw new Error('unreachable');
-          }
+        plugin.setup(coreSetup);
+        plugin.start(coreStart).then(({ refresh, license$ }) => {
```

Comment:
Another case where errors are not going to bubble up correctly because the errors aren't being caught and passed to done.

Reply:
The test is now async: https://github.com/elastic/kibana/pull/78720/commits/1116cefa6ae6f9bc1cb3d8aae9c1801d24a10a4f#diff-7aff575c26821dff90d1dcc084b8a345